crates/rpc/engine/mod.rs

```diff
@@ -85,6 +85,26 @@ pub fn new_payload_v3(
             "Invalid blob_versioned_hashes",
         ));
     }
+    // Check that the incoming block extends the current chain
+    let last_block_number = storage
+        .get_latest_block_number()
+        .map_err(|_| RpcErr::Internal)?;
+    if let Some(latest) = last_block_number {
```

Comment:
there should _always_ be a last_block_number, right?

Reply:
Fixed